File: 11기 코딩테스트_230308/연습1/DecimalToHexadecimal/DecimalToHexadecimal_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.DecimalToHexadecimal.DecimalToHexadecimal import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        n = 239

        # when
        result = solution(n)

        # then
        self.assertEqual('ef', result)

    def test_2(self):
        # given
        n = 1

        # when
        result = solution(n)

        # then
        self.assertEqual('1', result)

    def test_3(self):
        # given
        n = 292184

        # when
        result = solution(n)

        # then
        self.assertEqual('47558', result)

    def test_4(self):
        # given
        n = 94328293

        # when
        result = solution(n)

        # then
        self.assertEqual('59f55e5', result)

    def test_5(self):
        # given
        n = 3426

        # when
        result = solution(n)

        # then
        self.assertEqual('d62', result)

    def test_6(self):
        # given
        n = 842357182

        # when
        result = solution(n)

        # then
        self.assertEqual('323559be', result)


if __name__ == '__main__':
    unittest.main()

File: 11기 코딩테스트_230308/연습1/DecimalToHexadecimal/DecimalToHexadecimal_test.js
import assert from 'assert'
import solution from './DecimalToHexadecimal.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const n = 239

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('ef', result)
    done();
  });

  it('test 2', function (done) {
    // given
    const n = 1

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('1', result)
    done();
  });

  it('test 3', function (done) {
    // given
    const n = 292184

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('47558', result)
    done();
  });

  it('test 4', function (done) {
    // given
    const n = 94328293

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('59f55e5', result)
    done();
  });

  it('test 5', function (done) {
    // given
    const n = 3426

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('d62', result)
    done();
  });

  it('test 6', function (done) {
    // given
    const n = 842357182

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('323559be', result)
    done();
  });
})
